### PR TITLE
Product entity

### PR DESCRIPTION
**Pull request by Maciek2k3**
Created product entity, since is connected with Cart/Groups/OrderItem, I've created basic class only for correlations, if they are working properly

### CODE REVIEW

**Review by Ktubbo — COMMENTED** [review]
I think that ajax in frontend will be looking for those names in Json.

> {
>     "id": 1,
>     "name": "kurtka zimowa",
>     "description": "Pellentesque (...) purus.",
>     "price": 100,
>     "groupId": "1"
> }

**Comment by Maciek2k3** [comment]
Yes, agree, that's why we would have dto like this:
   private Long id;
    private String productName;
    private String productDescription;
    private Double price;
  For my opinion we need to have correlated entity acc. to EDR

**Comment by Ktubbo** [comment]
I think that those:

```
@Column(name = "productId")
@Column(name = "productName")
@Column(name = "productDescription")
```
should be named like this:

```
{
"id": 1,
"name": "kurtka zimowa",
"description": "Pellentesque (...) purus.",
"price": 100,
"groupId": "1"
}
```

but I could be wrong.

EDIT:
Or maybe those should be named differently (it is possible to map json by `@JsonProperty` annotation)
```
private String productName;
private String productDescription;
```

**Comment by Maciek2k3** [comment]
Yes, sorry you are correct, I will update ASAP

File: src/main/java/com/kodilla/ecommercee/domian/Product.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Column(name = "price")
+    @NotNull
+    private Double price;
+
+    @ManyToMany(mappedBy = "productList")
+    private List<Cart> cartList;
+
+    @ManyToOne
+    @JoinColumn(name = "groupId")
+    private Groups groups;
```
Comment (Ktubbo) [inline src/main/java/com/kodilla/ecommercee/domian/Product.java]: I think that this class should be named `Group`.

**Review by Ktubbo — APPROVED** [review]

File: src/main/java/com/kodilla/ecommercee/domian/Product.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @Column(name = "name")
+    @NotNull
+    private String productName;
+
+    @Column(name = "description")
+    private String productDescription;
```
Comment (Bauaser-kun) [inline src/main/java/com/kodilla/ecommercee/domian/Product.java:30]: Shouldn't we ensure that the product will always have some kind of description? (Adding @ NotNull annotation)?

Comment (Ktubbo) [reply src/main/java/com/kodilla/ecommercee/domian/Product.java:30]: I will not agree. I think there should be possibility to have empty description.

**Review by Ktubbo — DISMISSED** [review]

File: src/main/java/com/kodilla/ecommercee/domian/Product.java (hunk 2 above)
Comment (Bauaser-kun) [reply src/main/java/com/kodilla/ecommercee/domian/Product.java:30]: Ok. It was just a suggestion anyway, no changes requested

**Review by Bauaser-kun — CHANGES_REQUESTED** [review]
Have you created repository?
https://snipboard.io/3yYfvz.jpg

**Review by Ktubbo — APPROVED** [review]

**Review by Bauaser-kun — APPROVED** [review]

**Pull request merged**